cpp/test/neighbors/brute_force_c.cu

```diff
@@ -135,3 +406,79 @@ TEST(BruteForceC, BuildSearch)
   cudaFree(neighbors_data);
   cudaFree(distances_data);
 }
+
+TEST(BruteForceC, BuildSearchWithFilter)
+{
+  int64_t n_rows       = 8096;
+  int64_t n_queries    = 128;
+  int64_t n_dim        = 32;
+  uint32_t n_neighbors = 8;
+
+  float sparsity   = 0.2;
+  int64_t n_filter = (n_queries * n_rows + 31) / 32;
+  std::vector<uint32_t> filter_h(n_filter);
+  int64_t nnz = create_sparse_matrix(n_queries, n_rows, sparsity, filter_h);
+
+  cuvsDistanceType metric = L2Expanded;
+  bool select_min         = cuvs::distance::is_min_close(metric);
+
+  std::vector<float> distances_ref_h(
+    n_queries * n_neighbors,
+    select_min ? std::numeric_limits<float>::infinity() : std::numeric_limits<float>::lowest());
+  std::vector<int64_t> neighbors_ref_h(n_queries * n_neighbors, static_cast<int64_t>(0));
+
+  float *index_data, *query_data, *distances_data;
+  int64_t* neighbors_data;
+  uint32_t* filter_data;
+  cudaMalloc(&index_data, sizeof(float) * n_rows * n_dim);
```

Comment:
Can we please use the C APIs that we've created for RMM instead of using cudaMalloc directly? 

If cudaMalloc is being used in other C-based gtests then we need to fix those also- if that's the case, can you create an issue to clean up the others too?

Reply:
I will directly change all of the `cudaMalloc` under the directory of `neighbors` in this issue.

Reply:
thanks for making this change @rhdong - I appreciate that you fixed the other c-api unittests as well